src/common/function/src/scalars/json/json_get.rs

```diff
@@ -172,41 +175,211 @@ impl Function for JsonGetString {
         Self::NAME
     }
 
-    fn return_type(&self, _: &[DataType]) -> datafusion_common::Result<DataType> {
+    fn return_type(&self, _: &[DataType]) -> Result<DataType> {
         Ok(DataType::Utf8View)
     }
 
     fn signature(&self) -> &Signature {
         &self.signature
     }
 
-    fn invoke_with_args(
-        &self,
-        args: ScalarFunctionArgs,
-    ) -> datafusion_common::Result<ColumnarValue> {
+    fn invoke_with_args(&self, args: ScalarFunctionArgs) -> Result<ColumnarValue> {
         let [arg0, arg1] = extract_args(self.name(), &args)?;
-        let arg0 = compute::cast(&arg0, &DataType::BinaryView)?;
-        let jsons = arg0.as_binary_view();
+
         let arg1 = compute::cast(&arg1, &DataType::Utf8View)?;
         let paths = arg1.as_string_view();
 
-        let size = jsons.len();
-        let mut builder = StringViewBuilder::with_capacity(size);
+        let result = match arg0.data_type() {
+            DataType::Binary | DataType::LargeBinary | DataType::BinaryView => {
+                let arg0 = compute::cast(&arg0, &DataType::BinaryView)?;
+                let jsons = arg0.as_binary_view();
+                jsonb_get_string(jsons, paths)?
+            }
+            DataType::Struct(_) => {
+                let jsons = arg0.as_struct();
+                json_struct_get_string(jsons, paths)?
+            }
```

Comment:
Should we treat nonexistence or wrong type as NULL instead of error?

Reply:
this wild card branch is data type error I think it should be good. postgres will also raise error when data type mismatch.

Reply:
nonexistence is treated as null when iterating the array

wrong type is better treated as error to fail fast (less surprise)